src/components/PopUpWithForm.js

```diff
@@ -7,17 +7,36 @@ export default class PopUpWithForm extends PopUp {
     this._handleFormSubmit = handleFormSubmit;
     this._config = config;
     this._form = this._popup.querySelector(this._config.formSelector);
-    this.inputEl = this._popUpElement.querySelectorAll(".modal__input");
-    this.inputValues = document.querySelectorAll(this._config.inputSelector);
+    this.inputEl = this._popup.querySelectorAll(".modal__input");
+    this.inputValues = {};
+    this._submitButton = this._popup.querySelector(".modal__button");
+  }
+
+  setLoading(isLoading) {
+    if (isLoading) {
+      this._submitButton.textContent = "Saving...";
+    } else {
+      this._submitButton.textContent = "Save";
+    }
   }
 
   setEventListeners() {
     super.setEventListeners();
-    this._form.addEventListener("submit", (e) => {
-      e.preventDefault();
-      this._handleFormSubmit(this._getInputValues());
-      this._form.reset();
-    });
+    if (this._form) {
+      this._form.addEventListener("submit", (e) => {
+        e.preventDefault();
+        this._handleFormSubmit(this._getInputValues())
+          .then(() => {
+            this._form.reset();
+            this.close();
+          })
+          .catch((err) => {
+            console.error(err);
+          });
```

Comment:
* There is an error when I add a new card or change the avatar https://snipboard.io/nXEWBb.jpg 


Please, delete `then` and `catch` from `PopUpWithForm`. You don't need them here.


You should make a special method `resetForm` for resetting to call it in `index.js` in a `then` block


```js
  resetForm() {
    this._form.reset();
  }


```